slither/slithir/convert.py

```diff
@@ -837,6 +843,9 @@ def extract_tmp_call(ins: TmpCall, contract: Optional[Contract]):  # pylint: dis
                 ins.lvalue,
                 ins.type_call,
             )
+            custom_error_sym = resolve_error(str(ins.ori.variable_right), ins.ori.variable_left)
```

Comment:
On line 996, there is a `CustomError` case handler (presumably for top-level custom error calls).
```solidity
    if isinstance(ins.called, CustomError):
        sol_function = SolidityCustomRevert(ins.called)
        s = SolidityCall(sol_function, ins.nbr_arguments, ins.lvalue, ins.type_call)
        s.set_expression(ins.expression)
        return s
```

Our implementation should reflect the top-level path's `CustomError` handling.
(Though, it's annoying that they are still using `SolidityCall`, instead of creating `CustomErrorCall` there.)

Also, I'd like this code to be more symmetric to the `EventCall` case above.
Let's create a separate if-statement here just like the event_sym leaving this default case as the normal library call.

Reply:
Will do.

Reply:
I guess I should clarify: the normal custom error revert already stored the instance of its custom error, this PR specifically want to address the case where a custom error from library is reverted.

Reply:
Ok, let's keep it a LibraryCall when a library error is constructed :(
So, our detector will have to deal with both LibraryCall and SolidityCall.

Reply:
That indeed is how the detector was written.